crossbar/worker/process.py

```diff
@@ -40,7 +40,7 @@
 __all__ = ('run',)
 
 
-def get_argument_parser(parser):
+def get_argument_parser(parser=argparse.ArgumentParser()):
     parser.add_argument('--reactor',
```

Comment:
No, we shouldn't do this: this default will only be instantiated ONCE, which is not what you intend I don't think. Instead, say `parser=None` and in the function itself `if parser is None: parser = argparse.ArgumentParser()`; this way you'll get a new parser each time `get_argument_parser()` is called -- it's probably only called once here, but..

Reply:
If this function is called without an argument, a new parser is created (old logic, that was happening previously as well, though inside run()) and if we provide a parser as an argument, it uses that, allowing code reusability for cli.py

Reply:
No, a new parser is only created ONCE, at import time. So if you call it a second time, you'll get the same parser again.